mark.c

```diff
@@ -810,23 +816,38 @@ GC_INNER mse * GC_mark_from(mse *mark_stack_top, mse *mark_stack,
         for(;;) {
           PREFETCH(limit - PREF_DIST*CACHE_LINE_SIZE);
           GC_ASSERT((word)limit >= (word)current_p);
-          deferred = *(word *)limit;
```

Comment:
Rather than deleting this code, should it be guarded by `#ifdef RISCV && CHERI` or maybe just `#ifdef CHERI`?

Reply:
fixed in commit [408f667b](https://github.com/capablevms/bdwgc/pull/1/commits/408f667bb59d4d389e44ff70d5631bf922413ed9)